### PR TITLE
Document jax.experimental.multihost_utils

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: docs/jax.experimental.multihost_utils.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Multihost Utils API Reference
+-----------------------------
+.. autosummary::
+   :toctree: _autosummary
+
+   broadcast_one_to_all
+   sync_global_devices
+   process_allgather
+   assert_equal
+   reached_preemption_sync_point
```
Comment (yashk2810) [inline docs/jax.experimental.multihost_utils.rst:17]: Please remove this

Comment (nouiz) [reply docs/jax.experimental.multihost_utils.rst]: done

File: jax/experimental/multihost_utils.py [hunk 2]
```diff
@@ -357,36 +353,30 @@ def global_array_to_host_local_array_impl(
...
-
-  with mesh:
-    global_out = pjitted_fun(global_inputs)
-
-  host_local_output = multihost_utils.global_array_to_host_local_array(
-    global_out, mesh, out_pspecs)
-  ```
+  >> from jax.experimental import multihost_utils
+  >>
+  >> global_inputs = multihost_utils.host_local_array_to_global_array(host_local_inputs, global_mesh, in_pspecs)
```
Comment (hawkinsp) [inline jax/experimental/multihost_utils.py:369]: It should be sufficient to suffix each of these lines with `# doctest: +SKIP`

File: jax/experimental/multihost_utils.py [hunk 3]
```diff
@@ -257,35 +257,31 @@ def host_local_array_to_global_array_impl(
...
-    host_local_inputs, global_mesh, in_pspecs)
-
-  with mesh:
-    global_out = pjitted_fun(global_inputs)
-
-  host_local_output = multihost_utils.global_array_to_host_local_array(
-    global_out, mesh, out_pspecs)
+  >>> from jax.experimental import multihost_utils
+  >>>
+  >>> global_inputs = multihost_utils.host_local_array_to_global_array(host_local_inputs, global_mesh, in_pspecs)
```
Comment (hawkinsp) [inline jax/experimental/multihost_utils.py:273]: Same here.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**